### PR TITLE
libsais 2.7.1 (new formula)

### PR DESCRIPTION
**Pull request by Atgoogat**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/libsais.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  desc "Fast linear time suffix array, lcp array and bwt construction"
+  homepage "https://github.com/IlyaGrebnov/libsais"
+  url "https://github.com/IlyaGrebnov/libsais.git",
+    tag:      "v2.7.1",
+    revision: "c0b2bba165ae759fcc034d8f22d973cea8d7329c"
+  license "Apache-2.0"
+  head "https://github.com/IlyaGrebnov/libsais.git", branch: "master"
+
+  depends_on "gcc" => :build
+  depends_on "make" => :build
```
Comment (chenrui333) [inline Formula/libsais.rb:11]:
```suggestion
```

File: Formula/libsais.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+class Libsais < Formula
+  desc "Fast linear time suffix array, lcp array and bwt construction"
+  homepage "https://github.com/IlyaGrebnov/libsais"
+  url "https://github.com/IlyaGrebnov/libsais.git",
+    tag:      "v2.7.1",
+    revision: "c0b2bba165ae759fcc034d8f22d973cea8d7329c"
```
Comment (chenrui333) [inline Formula/libsais.rb:6]: can we try this?
```suggestion
  url "https://github.com/IlyaGrebnov/libsais/archive/refs/tags/v2.7.1.tar.gz"
  sha256 "5f459ad90cd007c30aaefb7d122bba2a4307ea02915c56381be4b331cca92545"
```

File: Formula/libsais.rb (hunk 1 above)
Comment (Atgoogat) [reply Formula/libsais.rb:11]: Why "make" should not be listed as a dependency?

File: Formula/libsais.rb (hunk 2 above)
Comment (Atgoogat) [reply Formula/libsais.rb:6]: Works fine! :+1:

**Comment by chenrui333** [comment]
still does not build

```
gcc -Wall -O2 -shared -Wl,-soname,libsais.so.2 src/libsais.o -o libsais.so.2
ld: unknown option: -soname
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Atgoogat** [comment]
That seems to be a problem related to usage of clang. As the option "soname" is there not supported. With gcc it works fine.
Is there a way to handle this?

**Comment by Atgoogat** [comment]
Okay the error is fixed now. The dynamic lib will not be build anymore.
This is not optimal, but for other solutions a fix of the Makefile is required (or at least thats what I think).

**Comment by chenrui333** [comment]
> This is not optimal, but for other solutions a fix of the Makefile is required (or at least thats what I think).

does that mean this PR is still in progress?

**Comment by Atgoogat** [comment]
Sorry for the confusion.
Yes in fact I would consider it done and ready for merge at the moment.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@Atgoogat, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**